pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -843,7 +814,66 @@ func (h *AuthorizationHandler) validatePreAuthenticatedURLTokenRequest(
 }
 
 // nolint:gocognit
-func (h *AuthorizationHandler) validateRequest(
+func (h *AuthorizationHandler) ValidateRequestWithoutTx(
+	ctx context.Context,
+	r protocol.AuthorizationRequest,
+) (*AuthorizationParams, *AuthorizationResultError) {
```

Comment:
Change the return type to `(context.Context, *AuthorizationParams, *AuthorizationResultError)`.

The semantics is: we always return `ctx` even in case of error. So the caller can always write something like

```go
ctx := ...
ctx, params, err := ValidateRequestWithoutTx(ctx, ...)
if err != nil {
  // Handle error. ctx should be unmodified
}
// ctx should be modified here.
```